subsys/shell/Kconfig.backends

```diff
@@ -18,9 +18,39 @@ config SHELL_BACKEND_SERIAL
 	bool "Enable serial backends."
 	default y
 	select SERIAL
+	select RING_BUFFER
 	help
 	  Enable serial backends.
 
+# Internal config to enable UART interrupts if supported.
+config SHELL_BACKEND_SERIAL_FORCE_INTERRUPTS
+	bool
+	default y
+	depends on SERIAL_SUPPORT_INTERRUPT
+	select UART_INTERRUPT_DRIVEN
+
+if SHELL_BACKEND_SERIAL
+
+config SHELL_BACKEND_SERIAL_TX_RING_BUFFER_SIZE
+	int "Set TX ring buffer size"
+	default 16
+	depends on UART_INTERRUPT_DRIVEN
+	help
+	  If UART is utilizing DMA transfers then increasing ring buffer size
+	  increases transfers length and reduces number of interrupts.
+
+config SHELL_BACKEND_SERIAL_RX_RING_BUFFER_SIZE
+	int "Set RX ring buffer size"
+	default 8
```

Comment:
That looks too low for me. I still didn't try to do the usual smoke test of pasting a decent length of line into the new shell - afraid of unpleasant surprises ;-). But with such a receive buffer size surprises are guaranteed. I'd suggest to set to 64 or something. On the contrary, tx buffer can decreased (if really needed.

Reply:
> On the other hand if user is only typing then 8 is more then enough and 56 bytes of ram on some platforms means another feature.

Well, we need to think about UX and common usecases. But granted, that can be seen as requiring separate consideration (and testing), so can be done later.